pdc/apps/release/views.py

```diff
@@ -1013,3 +1014,134 @@ def list(self, request, *args, **kwargs):
         %(SERIALIZER)s
         """
         return super(ReleaseVariantTypeViewSet, self).list(request, *args, **kwargs)
+
+
+class ReleaseGroupTypeViewSet(StrictQueryParamMixin,
+                              mixins.ListModelMixin,
+                              viewsets.GenericViewSet):
+    """
+    API endpoint that allows release-group types to be viewed.
+    """
+    serializer_class = ReleaseGroupTypeSerializer
+    queryset = models.ReleaseGroupType.objects.all()
+
+    def list(self, *args, **kwargs):
+        """
+        __Method__: GET
+
+        __URL__: $LINK:releasegrouptypes-list$
+
+        __Response__:
+
+        %(SERIALIZER)s
+        """
+        return super(ReleaseGroupTypeViewSet, self).list(*args, **kwargs)
+
+
+class ReleaseGroupsViewSet(ChangeSetModelMixin,
+                           StrictQueryParamMixin,
+                           viewsets.GenericViewSet):
+    """
+    API endpoint that allows release_group_types to be viewed or edited.
+    """
+
+    queryset = models.ReleaseGroup.objects.all()
+    serializer_class = ReleaseGroupSerializer
+    lookup_field = 'name'
+    lookup_value_regex = '[^/]+'
+    filter_class = filters.ReleaseGroupFilter
+
+    def create(self, request, *args, **kwargs):
+        """
+        __Method__: POST
+
+        __URL__: $LINK:releasegroups-list$
+
+        __Data__:
+
+        %(WRITABLE_SERIALIZER)s
+
+        __Response__:
+
+        %(SERIALIZER)s
+        """
+        data = request.data
+        keys = set(['description', 'name', 'type'])
+        arg_filter_map = ['active', 'releases']
+        allowed_keys = list(keys) + arg_filter_map
+        extra_keys = set(data.keys()) - set(allowed_keys)
+        if extra_keys:
+            return Response({'detail': '%s keys are not allowed' % list(extra_keys)},
+                            status=status.HTTP_400_BAD_REQUEST)
```

Comment:
thank you,  `keys ` and `arg_filter_map` is not need here, I will remove it.
BTW,  I consider these logic checking is prepare for QE, which can return a clear info as we want.